File: Game.c
// This is written in C

#include<stdio.h>
#include<stdlib.h>
#include<time.h>

int my_run=0,cpu_run=0, num, cpu; // run to add score, num for user input cpu for cpu input

void userBatting(int n){
    srand(time(0)); // to generate random number in real time
    cpu=rand()%10; // %10 to generate between 0 to 9

    if(n==cpu){
        printf("User is out!! It's cpu's batting\n");
        cpuBatting(-1); //passing mine own random value :)
    }
    else{
        my_run+=n;
        printf("Cpu's number is : %d\n", cpu);
        scanf("%d", &n);
        printf("User number is : %d\n", n);
        userBatting(n);
    }
}

void cpuBatting(int n){
    srand(time(0));
    cpu=rand()%10;

    if(n==cpu){
        printf("Cpu is out!!\n");
        game(); // to show who won
    }

    else{
        cpu_run+=cpu;
        printf("Cpu number is : %d\n", cpu);
        scanf("%d", &n);
        printf("User number is : %d\n", n);
        cpuBatting(n);
    }
}

void game(){
    printf("User run = : %d\n", my_run);
    printf("Cpu run = : %d\n", cpu_run);
    if(my_run>cpu_run)
    printf("User won!!!\n");
    else
    printf("Cpu won!!!\n");
}

int main(){
    printf("Enter your number\n"); // user will bat first always
    scanf("%d", &num);
    userBatting(num); // function for user batting
}

File: game.java
package Game;
import java.util.*;
public class handCricket {
    static Scanner sc=new Scanner(System.in);
    static int cpu,my_sum=0,cpu_sum=0;
    public static void main(String[] args) {

        System.out.println("Enter your number");
        int num=sc.nextInt();

        yourBatting(num);
    }
    static Random rn = new Random();
    public static void yourBatting(int num){
        cpu=rn.nextInt(6)+1;
        if(num==cpu){
            System.out.println("You're out!! Now Cpu's turn....");
            cpuBatting(-1);
        }
        else {
            my_sum+=num;
            System.out.println(cpu);
            num=sc.nextInt();
            yourBatting(num);
        }
    }
    public static void cpuBatting(int num){
        cpu=rn.nextInt(6)+1;
        if(num==cpu){
            System.out.println("Cpu is out....");
            game();
        }
        else {
            cpu_sum+=cpu;
            System.out.println(cpu);
            num=sc.nextInt();
            cpuBatting(num);
        }
    }
    public static void game(){
        if(my_sum>cpu_sum)
            System.out.println("You won, your total score is = " + my_sum);
        else
            System.out.println("Cpu won, cpu total score is = " + cpu_sum);
    }
}
